pkg/apiutil/serverapi/middleware.go

```diff
@@ -40,32 +42,77 @@ const (
 
 var initHTTPClientOnce sync.Once
 
+// dialClient used to dial http request.
+var dialClient = &http.Client{
+	Transport: &http.Transport{
+		DisableKeepAlives: true,
+	},
+}
+
+type runtimeServiceAuth struct {
+	s     *server.Server
+	group server.APIGroup
+}
+
+// NewRuntimeServiceAuth checks if the path is invalid.
+func NewRuntimeServiceAuth(s *server.Server, group server.APIGroup) negroni.Handler {
+	return &runtimeServiceAuth{s: s, group: group}
+}
+
+func (h *runtimeServiceAuth) ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	if IsServiceAllowed(h.s, h.group) {
+		next(w, r)
+		return
+	}
+
+	http.Error(w, "no service", http.StatusServiceUnavailable)
+}
+
+// IsServiceAllowed checks the service through the path.
+func IsServiceAllowed(s *server.Server, group server.APIGroup) bool {
+	opt := s.GetServerOption()
+	cfg := opt.LoadPDServerConfig()
+	if cfg != nil {
+		for _, allow := range cfg.RuntimeServices {
+			if len(allow) != 0 && group.Name == allow {
```

Comment:
`allow != ""`? I think we don't need it if `RuntimeService` initialized as `[]string{}`.